examples/simple/igraph_assortativity.c

```diff
@@ -0,0 +1,32 @@
+/* -*- mode: C -*-  */
+
+#include <igraph.h>
+#include <stdio.h>
+
+int main(){
+    igraph_t g;
+    igraph_vector_t types;
+    igraph_real_t assortativity;
+
+/* Create directed graph */
+    igraph_small(&g, 10, IGRAPH_DIRECTED, 
+                 0, 1, 0, 2, 0, 3, 0, 5,
+                 3, 6,
+                 4, 5,
+                 6, 4, 6, 7,
+                 7, 8, 7, 9,
+                 8, 9,
+                 9, 0,
+                 -1);
+
+    igraph_vector_init(&types, 0);
+    igraph_degree(&g, &types, igraph_vss_all(), IGRAPH_ALL, /* No self-loops */ 0);
```

Comment:
It seems to me that in your example, excluding self-loops does not serve any purpose. If you check the "time complexity" section of the `igraph_degree` documentation, you will see that removing self-loops has a performance cost.  It's best for examples to illustrate good practices (even when performance doesn't matter _for this specific example_) so I suggest using `1`.

(You can also use `IGRAPH_LOOPS` instead of `1`, which is self-explanatory.)

Reply:
I have changed it to `IGRAPH_LOOPS`.

Reply:
Feel free to click "Resolve conversation" whenever you address one of these comments.